prime-router/src/main/kotlin/azure/SettingsFunctions.kt

```diff
@@ -330,9 +339,14 @@ open class BaseFunction(
         clazz: Class<T>,
         organizationName: String? = null
     ): HttpResponseMessage {
-        authenticateAdmin(request)
-            ?: return HttpUtilities.unauthorizedResponse(request, authenticationFailure)
-
+        val claims = AuthenticatedClaims.authenticate(request)
+        if (claims == null || !claims.authorized(
+                setOf(PRIME_ADMIN_PATTERN, "$settingName.*.admin", "$settingName.*.user")
```

Comment:
```suggestion
                setOf(PRIME_ADMIN_PATTERN, "$organizationName.*.admin", "$organizationName.*.user")
```

Reply:
This actually does not work, functionality would fail and tests would fail with this suggested change

Reply:
So I get how this would work if the setting name was an org, but how does this work when fetching a receiver or sender?